0_intro/exercises/cat/src/main.rs

```diff
@@ -1,7 +1,14 @@
+use std::env;
+use std::io::Read;
+
 pub fn cat(path: &str) -> String {
-    todo!();
+    let mut file = std::fs::File::open(path).unwrap();
+    let mut contents = String::new();
+    file.read_to_string(&mut contents).unwrap();
+    contents
 }
 
 fn main() {
-    todo!();
+    let args: Vec<String> = env::args().collect();
```

Comment:
Iterators in Rust are lazy, so this is actually quite costly! You just want to access the first element, so you do not need to iterate over the whole iterator and collect it (but it's perfectly valid to do so). Also, what is often done is that, when using a single function from a module (in your case `std::env::args`), write a use statement for the function itself. Therefore,
```rs
use std::env;
// ...
env::args().collect();
```
would become
```rs
use std::env::args;
// ...
env::args().collect();
```

But your way is completely fine!